### ISSUE TITLE
When you use Az PowerShell to deploy deployment stack with Bicep template and parameters files you get warning/errors twice.

### ISSUE THREAD

**Issue by slavizh** [Azure/deployment-stacks#118]
**Describe the bug**
When you use Az PowerShell to deploy deployment stack with Bicep template and parameters files you get warning/errors twice.

![image](https://github.com/Azure/deployment-stacks/assets/11890856/72d47e28-fda5-4a86-8fa2-e48eda5faa36)

**To Reproduce**
Deployment of bicep template with bicep parameters file

**Expected behavior**
To show warnings once as  these warnings are for the template only.

**Screenshots**
provided

**Repro Environment**
Host OS: Linux Ubuntu 20.04
Powershell Version: 7.2

**Comment by dantedallag** [comment]
Is this just happening when the verbose flag is passed in? Would you mind sharing the exact command + parameters that led to this output?

**Comment by slavizh** [comment]
yeah with verbose. I have noticed same behavior with regular deployment commands. For me the best experience would be to have any warnings or errors once and if you have warnings on Bicep Parameters file and on Bicep template to have clear separation to know from there it is coming.

```
$deploymentStackParameters = @{
            Name                           = $deploymentStackName
            Description                    = $deploymentStackDescription
            Location                       = $deploymentLocation
            TemplateParameterFile          = $bicepParametersFullFilePath
            TemplateFile                   = $bicepMainTemplateFilePath
            DenySettingsMode               = $deploymentStackDenySettingsMode
            DenySettingsApplyToChildScopes = $deploymentStackDenySettingsApplyToChildScopes
            DenySettingsExcludedAction     = $deploymentStackDenySettingsExcludedActions
            DenySettingsExcludedPrincipal  = $deploymentStackDenySettingsExcludedPrincipals
            Force                          = $true
            Verbose                        = $true
            ErrorAction                    = 'SilentlyContinue'
            ErrorVariable                  = 'deploymentError'
        }
$deploymentOutput = New-AzSubscriptionDeploymentStack @deploymentStackParameters

```

It happens on any command as long as you are deploying  Bicep template with Bicep parameters file

**Comment by aperezmicrosoft** [comment]
Hi @slavizh - the duplicate output seen in verbose is by design. The reason for this is because the service expects a JSON file for both the Bicep and Bicepparam files, therefore every time the client compiles the BICEP into a JSON file, verbose output is produced. The client is not aware of whether the JSON file is the template or the parameter file, and therefore uses the same bicep module for conversion of both the template and parameter file.

**Comment by slavizh** [comment]
ok this is by design but for me this is not good experience to show the same warnings for the same files twice.

**Comment by torreymicrosoft** [comment]
Thanks @slavizh — revisiting this as part of an open-issue review for the ARM Deployments team.

To recap for anyone landing here: as @aperezmicrosoft explained back in August 2023, the doubled verbose warnings are by design today — the service consumes JSON for both the template and the parameters, so the client compiles the `.bicep` and the `.bicepparam` files separately, and since the same Bicep module handles both compiles, the template's warnings surface once per compile.

Your underlying ask is still a fair UX improvement: emit each warning once, and when warnings do come from both files, clearly label which file each one is from. 

I'll document this as a known limitation for now. I'll also leave the tagged `needs upvote` and feature request. 👍 on the issue helps us prioritize.
